### PR TITLE
Check if value is iterable before extend

### PR DESCRIPTION
**Pull request by achaussier**
Hello,
Without this test, raises an exception in this example :

    ...
    pre_tasks:
    roles:
      ...

### CODE REVIEW

File: lib/ansiblelint/utils.py [hunk 1]
```diff
@@ -272,7 +272,11 @@ def extract_from_list(blocks, candidates):
     for block in blocks:
         for candidate in candidates:
             if candidate in block:
-                results.extend(block[candidate])
+                if hasattr(block[candidate], '__iter__'):
```
Comment (willthames) [inline lib/ansiblelint/utils.py]: presumably `if isinstance(block[candidate], list)` would work equally well - do we expect anything other than a (possibly empty) list?

Comment (achaussier) [inline lib/ansiblelint/utils.py]: Hello,

Yes, you're right. I update the PR.

**Comment by achaussier** [comment]
Following your comment, I think It's better to return an error if key exists without a list value.

**Comment by phemmer** [comment]
Will this be customizable via a `-x` option? Having an empty list is syntactically valid.

**Comment by willthames** [comment]
@phemmer, I'd be happy to just ignore empty lists rather than raise an error. i.e. treat empty `pre_tasks` the same as no `pre_tasks`.

**Comment by achaussier** [comment]
I've updated to not raise an error if empty value

**Comment by willthames** [comment]
Can you squash the commits into one and then this one will be ready to go.

**Comment by achaussier** [comment]
Done

**Comment by willthames** [comment]
Version 2.3.6 includes this change.

**Pull request merged**